File: LeetCode/Python/110. Balanced Binary Tree.py
# type: ignore
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        
        self.balanced = True

        def DFS(root):
            if not root:
                return 0
            left = DFS(root.left)
            right = DFS(root.right)
            
            if abs(left - right) > 1:
                self.balanced = False
            return max(left, right) + 1
        DFS(root)
        return self.balanced
    

File: LeetCode/C#/110. Balanced Binary Tree.cs
using System;
public class Solution {
    public bool IsBalanced(TreeNode root) 
    { 
        bool balanced = true;

        DFS(root);
        return balanced;

        int DFS(TreeNode node)
        {
            if (node == null)
                return 0;
            int left = DFS(node.left);
            int right = DFS(node.right);

            if (Math.Abs(left - right) > 1)
                {
                    balanced = false;
                }
            return Math.Max(left,right) + 1;
        }
    }
}